### PR TITLE
feat(queries-preview): add isAnyQueryLoadedInPreview composable

### PR DESCRIPTION
**Pull request by ajperezbau**
EMP-1006

## Motivation and context

Add composable to check if any query passed to it is loaded in the query preview module.

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [ ] `Main`
- [x] Other. Specify: [feature/EMP-1006](https://github.com/empathyco/x/tree/feature/EMP-1006)

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-modules/queries-preview/composables/use-queries-preview.composable.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Composable helpers for the QueriesPreview module.
+ */
+interface UseQueriesPreview {
+  /**
+   * Checks if any of the queries passed have results in the queries previews.
+   *
+   * @param queries - The queries to check.
+   * @returns True if some query has results.
+   */
```
Comment (annacv) [inline packages/x-components/src/x-modules/queries-preview/composables/use-queries-preview.composable.ts:14]: This doc is from the code below 👀

File: packages/x-components/src/x-modules/queries-preview/composables/use-queries-preview.composable.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * Composable helpers for the QueriesPreview module.
+ */
+interface UseQueriesPreview {
+  /**
+   * Checks if any of the queries passed have results in the queries previews.
+   *
+   * @param queries - The queries to check.
+   * @returns True if some query has results.
```
Comment (annacv) [inline packages/x-components/src/x-modules/queries-preview/composables/use-queries-preview.composable.ts:13]:
```suggestion
   * Indicates if any of the queries previews have results.
```

File: packages/x-components/src/x-modules/queries-preview/composables/__tests__/use-queries-preview.composable.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        queriesPreview: {
+          queryWithResults: createQueryPreviewItem('queryWithResults'),
+          queryWithNoResults: createQueryPreviewItem('queryWithNoResults', {
+            totalResults: 0,
+            results: []
+          })
+        }
+      });
+      expect(isAnyQueryLoadedInPreview(['queryWithResults'])).toBe(true);
+    });
```
Comment (annacv) [inline packages/x-components/src/x-modules/queries-preview/composables/__tests__/use-queries-preview.composable.spec.ts:55]: Although it is a bit redundant, maybe we can split this in two tests and handle the noResults use case, too. But, up to you:
```suggestion
    it('returns true if any query has results', () => {
      resetXQueriesPreviewStateWith(store, {
        queriesPreview: {
          queryWithResults: createQueryPreviewItem('queryWithResults'),
          queryWithNoResults: createQueryPreviewItem('queryWithNoResults', {
            totalResults: 0,
            results: []
          })
        }
      });
      expect(isAnyQueryLoadedInPreview(['queryWithResults'])).toBe(true);
    });
    
     it('returns false if any query has no results', () => {
      resetXQueriesPreviewStateWith(store, {
        queriesPreview: {
          queryWithNoResults: createQueryPreviewItem('queryWithNoResults', {
            totalResults: 0,
            results: []
          })
        }
      });
      expect(isAnyQueryLoadedInPreview(['queryWithNoResults'])).toBe(false);
    });
```

**Review by annacv — DISMISSED** [review]
🎉

File: packages/x-components/src/x-modules/queries-preview/composables/__tests__/use-queries-preview.composable.spec.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  ) => QueryPreviewItem = (query, queryPreviewItem) => {
+    const results = getResultsStub();
+    return {
+      results: results,
+      totalResults: results.length,
+      status: 'success',
+      request: getQueryPreviewRequest(query),
+      ...queryPreviewItem
+    };
+  };
```
Comment (CachedaCodes) [inline packages/x-components/src/x-modules/queries-preview/composables/__tests__/use-queries-preview.composable.spec.ts:27]: Might be a good idea to make this a stub generator like the ones we have for results, NQ, etc, maybe move the `getQueryPreviewRequest` function there too.

**Review by CachedaCodes — CHANGES_REQUESTED** [review]

**Review by CachedaCodes — DISMISSED** [review]

**Review by CachedaCodes — APPROVED** [review]

**Pull request merged**